### PR TITLE
Create space from template

### PR DESCRIPTION
**Pull request by motechFR**

### CODE REVIEW

File: components/common/CreateSpaceForm/CreateSpaceForm.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+import Alert from '@mui/material/Alert';
+import Box from '@mui/material/Box';
+import Divider from '@mui/material/Divider';
+import Grid from '@mui/material/Grid';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import type { Space } from '@prisma/client';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
```
Comment (msadura) [inline components/common/CreateSpaceForm/CreateSpaceForm.tsx:14]: awestome to see react-hook-form, I've got the feeling we do too many forms "manually"

File: components/common/CreateSpaceForm/SpaceTemplateOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+type SelectNewSpaceTemplateProps = {
+  onSelect: (option: SpaceCreateTemplate) => void;
+  selectedTemplate: SpaceCreateTemplate;
+};
+
+const templates = typedKeys(spaceContentTemplates);
+
+export function SelectNewSpaceTemplate({ onSelect, selectedTemplate }: SelectNewSpaceTemplateProps) {
```
Comment (msadura) [inline components/common/CreateSpaceForm/SpaceTemplateOptions.tsx:56]: I would stick for 1 component per file. In this case it is short, but recently I worked with page sidebar navigation, where single file has a couple components and it was really hard to find stuff around.

**Review by msadura — APPROVED** [review]
Codewise looks awesome, tested and it worked, I have only small UI comments.

**Comment by msadura** [comment]
This modal would look better as "fullScreen" https://mui.com/material-ui/api/dialog/ to make more space for content
Screenshot 2023-01-27 at 12 46 03

**Comment by msadura** [comment]
This screen is imo is a bit chaotic UX wise (mix of different fonts, alignments and titles) and it is hard to understand "what is what", but this does not feel like a blocker (I will mention this on discord). 
One little thing tho - X icon does nothing

Screenshot 2023-01-27 at 12 49 44

File: components/common/CreateSpaceForm/SpaceTemplateOptions.tsx (hunk 2 above)
Comment (motechFR) [reply components/common/CreateSpaceForm/SpaceTemplateOptions.tsx:56]: Gotcha, small edit, can do!

File: pages/[domain]/settings/space.tsx [hunk 3]
```diff
@@ -28,8 +27,18 @@ import { setTitle } from 'hooks/usePageTitle';
 import { usePreventReload } from 'hooks/usePreventReload';
 import { useSpaces } from 'hooks/useSpaces';
 import { useUser } from 'hooks/useUser';
+import { defaultTemplate } from 'lib/spaces/utils';
 import isSpaceAdmin from 'lib/users/isSpaceAdmin';
 
+const schema = yup.object({
```
Comment (Devorein) [inline pages/[domain]/settings/space.tsx:33]: This is duplicated in another place. Its best to keep a single validation object :D

**Review by Devorein — APPROVED** [review]
Import from notion failed on my end (with the same environment variables as @motechFR) but its working just fine over there.

**Comment by motechFR** [comment]
@msadura Thanks for the feedback. I've created a fast follow mobile task to fix the create space form

**Pull request merged**